### PR TITLE
Add module_name variable

### LINKED ISSUE #1: Add separate input for pytest pyargs check vs package install name
As pointed out by @lilyminium in https://github.com/MDAnalysis/GridDataFormats/pull/126 we use inputs.package_name for both pip installing and pytest checking. So in some cases (for example GridDataFormats), that causes an issue where package_name != module_name.

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #1 

Probably need tests for griddata/formats too...

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
         ref: 'v0.1.0'
 
     - name: pypi
-      uses: IAlibay/pypi-deployment@main
```
Comment (IAlibay) [inline .github/workflows/ci.yaml:34]: Oof I have been asleep at the wheel on that one.. I can't believe how many mistakes I made here (although I guess it did just redirect).

File: action.yaml [hunk 2]
```diff
@@ -21,9 +21,13 @@ inputs:
     default: 'pytest'
     required: false
   package_name:
-    description: 'Name of package on PyPi, only needed if testing package post-upload'
+    description: 'Name of package on PyPi, only needed if testing package post-upload. Used for pip installing'
     default: 'MDAnalysis'
     required: false
+  module_name:
+    description: 'Name of package directory, only needed if testing package post-upload. Used for pytest pyargs'
```
Comment (IAlibay) [inline action.yaml]:
```suggestion
    description: 'Name of package directory, only needed if testing package post-upload. Used for pytest pyargs. If unset will use `package_name`'
```
Something like that?

Comment (IAlibay) [reply action.yaml]: Or we could make it an explicit requirement.

**Review by IAlibay — CHANGES_REQUESTED** [review]
Yeah we can probably just do a ref to an old tag of griddataformats, that should work fine.

File: action.yaml (hunk 2 above)
Comment (lilyminium) [reply action.yaml]: Up to you! I think this is a relatively uncommon scenario though, right?

Comment (IAlibay) [reply action.yaml]: Extra documentation never hurt anyone 😉

**Review by IAlibay — APPROVED** [review]
Odd, not sure why CI isn't getting triggered.

**Comment by IAlibay** [comment]
O no it is.. invalid workflow file.

**Comment by IAlibay** [comment]
Just undrafting to see if that's what is blocking the action.

**Comment by IAlibay** [comment]
I've added a sparse checkout to try to get out of a bit of a cyclic issue, but it turns out it still won't trigger on PR.

I'm going to merge this in the hope that there's some indication as to what's going on.

**Comment by IAlibay** [comment]
Took a bit of work and a new token but it works now, thanks @lilyminium !

**Pull request merged**